### ISSUE TITLE
Change the expires header (remove Dries birthday from Backdrop)

### ISSUE THREAD

**Issue by alanmels** [backdrop/backdrop-issues#4827]
> Should we re-purpose this issue to just change the expires header, or should we open a new issue for that?

I think a new issue might make sense. The issue description contains a lot of info we don't need for the small expires change and we shouldn't lose it either.

_Originally posted by @indigoxela in https://github.com/backdrop/backdrop-issues/issues/4817#issuecomment-750366824_

**Comment by alanmels** [comment]
Also copying the initial justification for the change request:

> As for the second one, for years I knew that one of the easiest way to check if a website is a Drupal or not, is to look at theexpires line of HTTP headers:
> 
> Sun, 19 Nov 1978 05:00:00 GMT
> 
> Unsurprisingly (as a fork) Backdrop has inherited the date and the hour when Dries Buytaert was born. I do respect the guy and know we - Backdrop users also - owe the guy, however I'd rather not pass the same trace to every Backdrop website out there. I would suggest to automatically change the expire line in the headers during the initial installation to some random number, so every Backdrop site had different expire information to make one less evident data gone. If for some reason a specific date is required then let's be inventive and change it to when Backdrop was born, or if Nate isn't against let's have his birthday there ;)

**Comment by BWPanda** [comment]
Either way, in greping core for '1978', I also found a few instances where Dries is mentioned by name:

```bash
$ grep 'Dries' core/
core/modules/simpletest/tests/common.test:    $javascript = backdrop_add_js(array('backdrop' => 'rocks', 'dries' => 280342800), 'setting');
core/modules/simpletest/tests/common.test:    $this->assertEqual(280342800, $javascript['settings']['data'][2]['dries'], 'JavaScript setting is set correctly.');
core/modules/simpletest/tests/theme.test:      'text' => "Dries' birthday",
core/modules/simpletest/tests/theme.test:    $this->assertEqual('<time datetime="1978-11-19T00:00:00+0000">Dries&#039; birthday</time>', theme('datetime', $variables));
core/modules/simpletest/tests/theme.test:      'text' => "Dries' birthday",
core/modules/simpletest/tests/theme.test:    $this->assertEqual('<time datetime="1978-11-19">Dries&#039; birthday</time>', theme('datetime', $variables));
core/modules/simpletest/tests/theme.test:      'text' => "<span>Dries' birthday</span>",
core/modules/simpletest/tests/theme.test:    $this->assertEqual('<time datetime="1978-11-19T00:00:00+0000"><span>Dries\' birthday</span></time>', theme('datetime', $variables));
```

Please let's change that too!

**Comment by alanmels** [comment]
So what date the community wants to change the date to? As I initially suggested there could be several options:

1. Random date during the installation of a Backdrop website. This would make every Backdrop website to have unique Expire date in Headers.
2. Have just found those lovely pictures on https://backdropcms.org/news/its-time-to-start-planning-your-backdrop-cms-birthday-party searching for the date. If I got it right, Backdrop CMS was born on Jan 15th, 2015, right?
3. I can say Nate is giving an impression not so egocentric guy (comparatively ...), however his idea of forking Drupal and making it real despite all the opposing voices in Drupal community has been proving right and bright. I believe it would be totally fair if Backdrop community decides to put his birthday in Expires line.

**Comment by BWPanda** [comment]
I'd say let's use Backdrop's birthday: https://github.com/backdrop/backdrop/releases/tag/v1.0.0

**Comment by alanmels** [comment]
So, it's 16th, not 15th? I wonder why celebrations take place on 15th?

**Comment by BWPanda** [comment]
Initial PR: https://github.com/backdrop/backdrop/pull/3456

I changed instances of 'Dries' so far, will update later with all instances of 1978 replaced too.

**Comment by BWPanda** [comment]
Because it was the 15th in the US timezone. But when changing simpletests to use a date/time, I'm not sure if where to set the timezone, so I'm using UTC to be safe...

**Comment by alanmels** [comment]
> I'd say let's use Backdrop's birthday: https://github.com/backdrop/backdrop/releases/tag/v1.0.0
> 
> Which was Fri, 16 Jan 2015 07:50am (UTC).

Nice, but will hear others on this? 

As a side note, I'd prefer Backdrop Community to come up with some kind of procedure to further define and strengthen its consensus principle. A voting system would do just fine to finalize some decisions.

**Comment by BWPanda** [comment]
Finished updating PR.

**Comment by alanmels** [comment]
> > I wonder why celebrations take place on 15th?
> 
> Because it was the 15th in the US timezone. 

Well, then maybe the 16th should be officially considered as Backdrop birth date and so the celebrations, because we are already working from several different countries and eventually the community will grow and cover more countries. So relying on UTC would be just right.

**Comment by jenlampton** [comment]
I'm in favor of using Backdrop's Birthday (Jan 15th, 2015) but I am not particular about what time or which timezone. What about midnight UTC? Or should we do something clever like 1:15am UTC (1/15 1:15)?

**Comment by stpaultim** [comment]
I'm happy with Birthday idea. Don't really care how we define the birthday (15th or 16th). 

The 15th coincides with release day (at least in the US). Celebrating on the 16th might be less stress. I think we should plan a virtual birthday party this year on either the 15th or 16th. Will raise that elsewhere.

**Comment by klonos** [comment]
> I'm happy with Birthday idea. Don't really care how we define the birthday (15th or 16th).

Ditto 👍 ...it could be 15th on the US, while we celebrate this on the 16th in EU or AU and elsewhere 🤷 ...I don't care much about when Backdrop was born ...just thankful it happened, ❤️ the community and everyone involved 🙂

**Comment by klonos** [comment]
...I haven't tested, but code looks good 👍

**Comment by indigoxela** [comment]
Hm. The expires header works, but many form tests are failing - also in the Tugboat sandbox.

![expires](https://user-images.githubusercontent.com/842066/103080604-739e3c80-45d6-11eb-8049-ffe9ce7b139b.png)

**Comment by indigoxela** [comment]
It seems that Zen.CI is in bad mood today. This time two unrelated tests failed. @BWPanda mind to turn it off and on again?

**Comment by BWPanda** [comment]
Tests are passing now, thanks to @indigoxela 😊

**Comment by alanmels** [comment]
I'm happy to see the Backdrop community is becoming more receptive of changes which make sense. Thanks to everyone!

**Comment by indigoxela** [comment]
I did a final test in the sandbox, the expires header still works for me. (expires: Fri, 16 Jan 2015 07:50:00 GMT)

**Comment by klonos** [comment]
Code LGTM 👍

**Comment by quicksketch** [comment]
Thanks folks! The PR at https://github.com/backdrop/backdrop/pull/3456 looks great! I merged into 1.x.

In theory this could affect some software that "fingerprints" CMS's by certain aspects (such as builtwith.com). However I think that's fairly unlikely to be an issue because there are a lot of other more obvious aspects that can be used for identifying, such as `X-Generator` HTTP header and the `<meta name="generator" content="Backdrop CMS (http://backdropcms.org)" />` HTML tag. If anything, it might help separate Backdrop from accidentally being identified as Drupal.

This also might not qualify as a "bug". As such I only merged this into 1.x for 1.18.0. Thanks everyone!
